localstack-core/localstack/services/cloudwatch/provider_v2.py

```diff
@@ -835,6 +831,84 @@ def _get_timestamp(input: dict):
             history = [h for h in history if (date := _get_timestamp(h)) and date <= end_date]
         return DescribeAlarmHistoryOutput(AlarmHistoryItems=history)
 
+    def _evaluate_composite_alarms(self, context: RequestContext, triggering_alarm):
```

Comment:
I had some doubts about the length of this function and its effects on readability. From one side in this regard it is similar to `set_alarm_state` one for metric alarms. From the other side it is tempting to extract some smaller functions, for example for evaluating each individual composite alarm (the contents of outer for loop), or processing alarm actions. That would also mean propagating parameters through the call stack, which may hurt maintainability in the future.

What is localstack project's preference in this regard?

Reply:
After a fresh look I extracted simpler methods as the original one was too error-prone, with several return statements that already brought an unwanted behavior.